### PR TITLE
Update kafka spring example to align with the spec

### PR DESCRIPTION
**Pull request by lmolkova**
The Kafka Spring example does not align with the general messaging spec
see https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1493#discussion_r1638468822

This PR updates it to:
- use proper parent/child relationships
- simplify the use case
- add settlement

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* [x] Change log entry added, according to the guidelines in [When to add a changelog entry](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md#when-to-add-a-changelog-entry).
  * If your PR does not need a change log, start the PR title with `[chore]`
* [x] [schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.

### CODE REVIEW

File: docs/messaging/kafka.md [hunk 1]
```diff
@@ -129,40 +129,49 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
...
-| `messaging.operation.type` |  |  | `"process"` |  | `"receive"` |
-| `messaging.client.id` |  | `"5"` | `"5"` | `"5"` | `"8"` |
-| `messaging.kafka.message.key` | `"myKey"` | `"myKey"` | `"myKey"` | `"anotherKey"` | `"anotherKey"` |
-| `messaging.kafka.consumer.group` |  | `"my-group"` | `"my-group"` |  | `"another-group"` |
-| `messaging.kafka.destination.partition` | `"1"` | `"1"` | `"1"` | `"3"` | `"3"` |
-| `messaging.kafka.message.offset` | `"12"` | `"12"` | `"12"` | `"32"` | `"32"` |
+| Field or Attribute | Span Send | Span Poll | Span Process | Span Commit |
+|-|-|-|-|-|
+| Span name | `"send T"` | `"poll T"` | `"process T"` | `"commit T"` |
+| Parent |  |  |  (optional) Span Publish  | Span Process |
```
Comment (joaopgrassi) [inline docs/messaging/kafka.md:164]:
```suggestion
| Parent |  |  |  (optional) Span Send| Span Process |
```
Or even better

```suggestion
| Parent |  |  |  (optional) `"send T"` | Span Process |
```

No?

Comment (joaopgrassi) [reply docs/messaging/kafka.md:164]: Or align the names with the diagram, so it's easy to follow both

Comment (lmolkova) [reply docs/messaging/kafka.md]: thanks, done!

File: docs/messaging/kafka.md [hunk 2]
```diff
@@ -127,41 +127,50 @@ If an intermediary broker is present, `service.name` and `peer.service` will not
...
+```mermaid
+flowchart LR;
+  subgraph PRODUCER
+  P[Span 'send T']
+  end
+  subgraph CONSUMER
+  direction TB
+  R1[Span 'poll T']
+  R2[Span 'process T']
+  R3[Span 'commit T']
```
Comment (pyohannes) [inline docs/messaging/kafka.md:146]:
```suggestion
  P[Span Send T]
  end
  subgraph CONSUMER
  direction TB
  R1[Span Poll T]
  R2[Span Process T]
  R3[Span Commit T]
```

That would be consistent with other examples. No strong opinion which approach to take, however, ideally we should be consistent.

**Review by pyohannes — APPROVED** [review]

**Pull request merged**